controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -184,15 +184,28 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.
 		if err != nil {
 			return ctrl.Result{}, err
 		}
+
+		externalEtcdReady, err := external.IsReady(externalEtcd)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if !externalEtcdReady {
+			log.Info("Managed external etcd is not ready yet, requeueing", "managedExternalEtcd", klog.KObj(externalEtcd))
+			return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+		}
+
 		endpoints, found, err := external.GetExternalEtcdEndpoints(externalEtcd)
 		if err != nil {
 			return ctrl.Result{}, errors.Wrapf(err, "failed to get endpoint field from %v", externalEtcd.GetName())
 		}
-		if !found {
-			log.Info("Etcd endpoints not available")
-			return ctrl.Result{Requeue: true}, nil
-		}
 		currentEtcdEndpoints := strings.Split(endpoints, ",")
+
+		if !found || areEndpointsEmpty(currentEtcdEndpoints) {
+			log.Info("Managed external etcd endpoints not available, requeueing")
+			return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+		}
```

Comment:
so this is just checking if the endpoints list is empty or not which is def an improvement but do we also wanna check if the endpoint count matches the expected etcd count?

Reply:
replicas is not part of the external etcd contract, so the kcp doesn't really know about it. Different implementations of the etcd provider can different specs. `status.ready` and status.endpoints` are part of the contract and that's why we can rely on them.

tbh, i don't expect the code to get to this branch ever. if the number of endpoints is not equal to the number of replicas, the `status.ready` should probably be false. This is just another level of protection to avoid rolling nodes that will fail.

Reply:
okay that makes sense! Thanks!